client/post-editor/editor-delete-post/index.jsx

```diff
@@ -42,6 +42,7 @@ const EditorDeletePost = React.createClass( {
 			}
 
 			if ( this.props.onTrashingPost ) {
+				this.setState( { isTrashing: false } );
 				this.props.onTrashingPost( error );
 			}
```

Comment:
Let's simplify this block a bit.  Clearly we always want to `setState( { isTrashing: false } )`, regardless of whether or not there was an error, so let's remove the `if ( error )` condition above.

I would also recommend renaming this method:  it's called `handleTrashingPost` but really it is called when the trash operation is finished.  Something like `postTrashedCallback` should be fine.

But really, this can just be an anonymous function since it is only used once, and the code reads more clearly that way anyway.  It can be an arrow function to get rid of the `.bind( this )`.

And also it looks like there is a bug.  What happens if the current user does not have permission to trash a post?  (How are they in the editor in the first place?)  We will `setState( { isTrashing: true } )` but we will never reset it.

So finally, we have:

```js
sendToTrash() {
    if ( ! utils.userCan( 'delete_post', this.props.post ) ) {
        return;
    }

    this.setState( { isTrashing: true } );

    actions.trash( this.props.post, error => {
        this.setState( { isTrashing: false } );

        if ( this.props.onTrashingPost ) {
            this.props.onTrashingPost( error );
        }
    } );
}
```